src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -374,62 +379,55 @@ SafeToPushdownUnionSubquery(PlannerRestrictionContext *plannerRestrictionContext
 
 
 /*
- * FindUnionAllVar finds the variable used in union all for the side that has
- * relationRteIndex as its index and the same varattno as the partition key of
- * the given relation with relationOid.
+ * RangeTableOffsetCompat returns the range table offset(in glob->finalrtable) for the appendRelInfo.
+ * For PG < 13 this is a no op.
  */
-static Var *
-FindUnionAllVar(PlannerInfo *root, List *appendRelList, Oid relationOid,
-				Index relationRteIndex, Index *partitionKeyIndex)
+static int
+RangeTableOffsetCompat(PlannerInfo *root, AppendRelInfo *appendRelInfo)
 {
-	ListCell *appendRelCell = NULL;
-	AppendRelInfo *targetAppendRelInfo = NULL;
-	AttrNumber childAttrNumber = 0;
-
-	*partitionKeyIndex = 0;
+	#if PG_VERSION_NUM >= PG_VERSION_13
+	int skipAmount = FindPreviousParentAmount(root->append_rel_list, appendRelInfo);
 
-	/* iterate on the queries that are part of UNION ALL subselects */
-	foreach(appendRelCell, appendRelList)
+	int i = 1;
+	for (; i < root->simple_rel_array_size; i++)
 	{
-		AppendRelInfo *appendRelInfo = (AppendRelInfo *) lfirst(appendRelCell);
-
-
-		int rtoffset = RangeTableOffsetCompat(root, appendRelInfo);
-
-		/*
-		 * We're only interested in the child rel that is equal to the
-		 * relation we're investigating.
-		 */
-		if (appendRelInfo->child_relid - rtoffset == relationRteIndex)
+		RangeTblEntry *rte = root->simple_rte_array[i];
+		if (rte->inh)
 		{
-			targetAppendRelInfo = appendRelInfo;
+			/*
+			 * we skip the previous parents because we want to find the offset
+			 * for the given append rel info
+			 */
+			if (skipAmount) {
```

Comment:
minor: Lets be more explicit: `if (skipParentCount  > 0)` and improve the alignment of curly braces